spec/section/annotations.md

```diff
@@ -1,16 +1,294 @@
+
+| Property                | Domain                | Range               |
+|-------------------------|-----------------------|---------------------|
+| `rml:structuralAnnotation`     | `rml:LogicalView` | `rml:StructuralAnnotation` |
+
+Following [=structural annotations=] MAY be defined:
+- [=Unique=] (`rml:UniqueAnnotation`)
+- [=ForeignKey=] (`rml:ForeignKeyAnnotation`)
+- [=NotNull=] (`rml:NotNullAnnotation`)
+- OtherFunctionalDependency
+- [=IriSafe=] (`rml:IriSafeAnnotation`)
+- (datatype also?)
+- [=PrimaryKey=] (`rml:PrimaryKeyAnnotation`)
+- [=Inclusion=]
+
+Intuitively, semantics for the annotations above is the same as for relational databases.
+
+All structural annotations of a logical view <i>lv</i> MUST have an <dfn>on fields</dfn> property (`rml:onFields`), linking the structural annotation to a list of field names occurring in <i>lv</i>. Intuitively, property [=on fields=] specifies the fields in <i>lv</i> that are involved by the structural annotation. The semantics of this involvement depends on the specific annotation.
+
+| Property                | Domain                | Range               |
+|-------------------------|-----------------------|---------------------|
+| `rml:onFields`             | `rml:PrimaryKeyAnnotation` | `rdf:List` |
+
+### Invariance Principle
+
+Differently from integrity constraints in databases, structural annotations are intended to be <i>annotations</i>. That is, the following invariant principle should be satisfied:
+
+<i>For any source instances, the RDF graph produced by the RML engine over an RML file with annotations, and the same file where annotations have been removed, MUST be the same.</i>
+
+RML engines might exploit structural annotations, as they could totally ignore them. It is responsibility of the user to make sure that the annotations provided are indeed correct. Sanity checks MAY be provided by the RML engines themselves, but this is not mandatory. Note that providing wrong annotations to an engine that takes into account for annotations, for instance for applying optimizations, could result in a violation of the invariance principle, with unpredictable results.
+
+### IriSafe
+
+An <dfn>IriSafe</dfn> structural annotation (`rml:IriSafeAnnotation`) [=on fields=] _F_ indicates that the content of each field in _F_ is [IRI safe](https://www.w3.org/TR/r2rml/#dfn-iri-safe), that is, each field in _F_ does not contain any character that is not in the [`iunreserved` production](http://tools.ietf.org/html/rfc3987#section-2.2) in [RFC3987](http://tools.ietf.org/html/rfc3987).
+
+### PrimaryKey
+
+The <dfn>PrimaryKey</dfn> structural annotation (`rml:PrimaryKeyAnnotation`) is analogous to the notion of primary key for databases. Specifically, a [=PrimaryKey=] annotation [=on fields=]  _(f1, ..., fn)_  imposes two conditions:
```

Comment:
I would skip the sentence referring to databases: 
A [=PrimaryKey=] annotation [=on fields=] _(f1, ..., fn)_ imposes two conditions:

Reply:
Ok, removed.